libs/testserver/server.go

```diff
@@ -17,15 +18,17 @@ type Server struct {
 
 	t testutil.TestingT
 
-	RecordRequests bool
+	RecordRequests        bool
+	IncludeRequestHeaders []string
 
 	Requests []Request
 }
 
 type Request struct {
-	Method string `json:"method"`
-	Path   string `json:"path"`
-	Body   any    `json:"body"`
+	Headers map[string]string `json:"headers,omitempty"`
```

Comment:
You could set this type to' http.Header` (https://pkg.go.dev/net/http#Header) and then filter only based on the keys. It would be good to know if there are duplicate headers.

This also removes the need for the `len(v) == 0` check below.